docs/content/concepts/partitions-schedules-sensors/sensors.mdx

```diff
@@ -183,6 +183,49 @@ def asset_a_and_b_sensor_with_skip_reason(context):
         )
 ```
 
+#### Triggering a partitioned run after upstream partitions are materialized
+
+The <PyObject object="MultiAssetSensorEvaluationContext" method="latest_materialization_records_by_partition"/> method fetches the latest materialization records by partition for a given asset. Using this method, you can trigger downstream partitioned runs when the corresponding partitions of upstream assets are materialized.
+
+The following example monitors two upstream daily-partitioned assets, kicking off materializations of the corresponding partition in the downstream daily-partitioned asset. Whenever a partition is replaced in `upstream_daily_1` and the same partition has previously been materialized in `upstream_daily_2` (and vice versa), the sensor yields a run request for the downstream asset.
+
+```python file=/concepts/partitions_schedules_sensors/sensors/sensors.py startafter=start_daily_partitioned_asset_with_two_upstreams endbefore=end_daily_partitioned_asset_with_two_upstreams
+@multi_asset_sensor(
+    asset_keys=[AssetKey("upstream_daily_1"), AssetKey("upstream_daily_2")],
+    job=downstream_daily_job,
+)
+def trigger_daily_asset_if_all_upstream_partitions_materialized(context):
+    run_requests_by_partition = {}
+    for (
+        partition,
+        materializations_by_asset,
+    ) in context.latest_materialization_records_by_partition_and_asset().items():
+        for asset_key, materialization in materializations_by_asset.items():
```

Comment:
I see. Is there a reason we prefer that behavior over behavior that only rematerializes if all parents are rematerialized? I could see the case for either, but smaller simpler example code seems like a good tiebreaker.

Reply:
I think the "and" case is actually more complex. So say:
- Downstream asset monitors upstreams A and B
- A materializes partitions 1 and 2, in that order
- B materializes partitions 3, 2, 1 in that order
The cursor can't actually be updated unless if the set of partitions in the first N materializations are the same for upstreams A and B. So in this case, until we get a partition 3 materialization in asset A, we can't advance the cursor for B (and thus can't advance the cursor for A)

Reply:
Oh man, this is gnarly.  Thinking through all this stuff makes me think about how easy it would be for a user to mess this up.

If we had infinite resources, I think we'd ideally store a cursor component for every (asset key, partition) tuple instead of just for every asset key, right? Would that make programming against this significantly simpler? If so, I wonder if there's some data structure that we could use to make this possible.

Reply:
For example, what if the cursor was a `Mapping[AssetKey, Sequence[StorageIdPartitionKeyRange]]`?  Where each `EventTimePartitionKeyRange` had `start_storage_id`, `end_storage_id`, `start_partition_key`, and `end_partition_key` fields.

`{AssetKey("asset1"): [EventTimePartitionKeyRange(1, 5, "2020-01-01", "2020-05-05")]}` would mean that our sensor has handled every event that has a storage_id between 1 and 5 that has a partition key between "2020-01-01" and "2020-05-05".